libs/language-server/src/lib/ast/model-util.ts

```diff
@@ -4,26 +4,47 @@
 
 import { strict as assert } from 'assert';
 
-import { AstNode, assertUnreachable } from 'langium';
+import { AstNode, Reference, assertUnreachable } from 'langium';
 
 // eslint-disable-next-line import/no-cycle
 import { getMetaInformation } from '../meta-information/meta-inf-registry';
 
 import {
   BinaryExpression,
   BlockDefinition,
+  CompositeBlocktypeDefinition,
   PipelineDefinition,
   UnaryExpression,
+  isCompositeBlocktypeDefinition,
 } from './generated/ast';
 import { PipeWrapper, createSemanticPipes } from './wrappers/pipe-wrapper';
 
 export function collectStartingBlocks(
-  pipeline: PipelineDefinition,
+  container: PipelineDefinition | CompositeBlocktypeDefinition,
 ): BlockDefinition[] {
+  // For composite blocks the first blocks of all pipelines are starting blocks as they have inputs
+  if (isCompositeBlocktypeDefinition(container)) {
+    const startingBlocks = container.pipes
+      .map((pipe) => pipe.blocks[0])
+      .map((blockRef: Reference<BlockDefinition> | undefined) => {
+        if (
+          blockRef &&
+          blockRef.ref &&
+          getMetaInformation(blockRef.ref.type) !== undefined
+        ) {
+          return blockRef.ref;
+        }
+        return undefined;
+      })
+      .filter(Boolean) as unknown as BlockDefinition[];
```

Comment:
Just as a hint: this issue is also solvable with a type guard instead, then we can get rid of the type assertion. Type assertions can introduce bugs on refactoring as they overwrite the TypeScript type system, the type guard solution would show an error if the type of `x` was refactored.

```suggestion
      .filter((x): x is BlockDefinition => x !== undefined);
```